Changelog updated for 3.2.0 release (#3023)
AzureAD/microsoft-identity-web: changelog.md

3.2.0
=========

> ### Breaking change

Reviewer: ```suggestion
- Updated to Microsoft.Identity.Abstractions 7.0.1
- Updated to Microsoft.IdentityModel.* 8.1.0
- Updated to Microsoft.Identity.Client 4.64.1
 
### New features
```

> - IDownstreamApi overloads take a JsonTypeInfo<T> parameter to enable source generated JSON deserialization.[#3017]

Reviewer 2: ```suggestion
- .In .NET 8 and above, IDownstreamApi overloads take a JsonTypeInfo<T> parameter to enable source generated JSON deserialization.[#3017]
```

### Bug fixes:

> - Azure region is used while creating application keys for caching application objects [#3002]

Reviewer 2: ```suggestion
- Azure region is used while creating application keys when the TokenAcquisition service caches application objects, and the TokenAcquirerFactory caches TokenAcquirer  [#3002]
```

3.2.0
=========
- Updated to Microsoft.Identity.Abstractions 7.0.1
- Updated to Microsoft.IdentityModel.* 8.1.0
- Updated to Microsoft.Identity.Client 4.64.1
 
### New features

> - .In .NET 8 and above, IDownstreamApi overloads take a JsonTypeInfo<T> parameter to enable source generated JSON deserialization.[#3017]

Reviewer: ```suggestion
- In .NET 8 and above, `IDownstreamApi` overloads take a `JsonTypeInfo<T>` parameter to enable source generated JSON deserialization. See [#3017] for details.
```

### Bug fixes:

> - Azure region is used while creating application keys when the TokenAcquisition service caches application objects, and the TokenAcquirerFactory caches TokenAcquirer  [#3002]

Reviewer: ```suggestion
- Azure region is used while creating application keys when the TokenAcquisition service caches application objects, and the TokenAcquirerFactory caches TokenAcquirer. See  [#3002] for details.
```

> 

Reviewer: ```suggestion
### Fundamentals:
- Improved test coverage for `GetCacheKey`. See PR [#3020](https://github.com/AzureAD/microsoft-identity-web/pull/3020) for details.
- Update to .NET 9-RC1. See issue [#3025](https://github.com/AzureAD/microsoft-identity-web/issues/3025) for details.
- Fix static analysis warnings. See PR [#3024](https://github.com/AzureAD/microsoft-identity-web/pull/3024) for details.
```

3.2.0
=========
- Updated to Microsoft.Identity.Abstractions 7.0.1
- Updated to Microsoft.IdentityModel.* 8.1.0
- Updated to Microsoft.Identity.Client 4.64.1
 
### New features

> - In .NET 8 and above, `IDownstreamApi` overloads take a `JsonTypeInfo<T>` parameter to enable source generated JSON deserialization. See [#3017] for details.

Reviewer: ```suggestion
- In .NET 8 and above, `IDownstreamApi` overloads take a `JsonTypeInfo<T>` parameter to enable source generated JSON deserialization. See issue [#2930] for details.
```

### Bug fixes:

> - Azure region is used while creating application keys when the TokenAcquisition service caches application objects, and the TokenAcquirerFactory caches TokenAcquirer. See  [#3002] for details.

Reviewer: ```suggestion
- Azure region is used while creating application keys when the TokenAcquisition service caches application objects, and the TokenAcquirerFactory caches TokenAcquirer. See [#3002] for details.
```
